src/Microsoft.IdentityModel.Protocols.WsFederation/WsFederationMetadataSerializer.cs

```diff
@@ -191,6 +195,9 @@ protected virtual SecurityTokenServiceTypeRoleDescriptor ReadSecurityTokenServic
             if (string.IsNullOrEmpty(roleDescriptor.TokenEndpoint))
                 LogHelper.LogWarning(LogMessages.IDX22807);
 
+            if (string.IsNullOrEmpty(roleDescriptor.ActiveTokenEndpoint))
```

Comment:
This implies that roleDescriptor.ActiveTokenEndpoint can be null or empty (not an error), but ReadSecurityTokenServiceEndpoint method throws LogMessages.IDX22812 exception if the element is empty.

Reply:
This is correct. I understand how this can be confusing. 

This inconsistency is there to avoid a breaking change ActiveTokenEndpoint is a new property that may not be used today by consumers of our library, if they chose to populate this class in any way but our metadata serializer this value may be null. So we cannot really enforce this new property without introducing a breaking change. 

The story is different for the configuration validator. This is a new class that no one is using so it can properly validate that the field exists and has the correct value.